sphinxcontrib/needs/directives/need.py

```diff
@@ -206,7 +206,7 @@ def get_sections_and_signature_and_needs(need_info):
         # Checking for a signature defined "above" the need.
         # Used and set normally by directives like automodule.
         # Only check as long as we haven't found a signature
-        if signature and current_node.parent and current_node.parent.children:
+        if not signature and current_node.parent and current_node.parent.children:
```

Comment:
Not sure, but does the code still detects signatures, if `not signature` is used?
I have the feeling it does not.

Reply:
must be `None`, as `signature` is initialized  with `none` and therefore this `if` is always False